src/parameters/update_cost_parameters.jl

```diff
@@ -114,30 +203,45 @@ function _update_pwl_cost_expression(
     return gen_cost
 end
 
+# For multi-start variables, we need to get a subset of the parameter
+_index_into_param(cost_data, ::T) where {T <: Union{StartVariable, MultiStartVariable}} =
+    start_up_cost(cost_data, T())
+_index_into_param(cost_data, ::VariableType) = cost_data
+
+# General case
+# (TODO this seemed rather decrepit before I got here and made big changes, make sure I didn't break anything)
 function update_variable_cost!(
     container::OptimizationContainer,
-    parameter_array::JuMPFloatArray,
-    parameter_multiplier::JuMPFloatArray,
-    attributes::CostFunctionAttributes{Float64},
-    component::T,
+    parameter_array::DenseAxisArray{T},
+    parameter_multiplier::JuMPFloatArray,  # TODO is the multiplier always `JuMPFloatArray`?
```

Comment:
I think so, the multiplier should be an array of floats, at least that is our intended purpose

Reply:
yes, multipliers are always floats